### ISSUE TITLE
`use_build_context_synchronously` is triggered when using `switch` statement

### ISSUE THREAD

**Issue by maRci002** [dart-lang/linter#4705]
**Describe the issue**

When using `BuildContext` inside a switch statement, if the first case awaits an async method, the second case triggers the `use_build_context_synchronously` lint rule. However, this is a false positive.

**To Reproduce**
```dart
import 'package:flutter/material.dart';

Future<void> test(BuildContext context) async {
  if (context.mounted) {
    switch (1) {
      case 1:
        await Future<void>.delayed(Duration.zero);
        break;
      case 2:
        Theme.of(context); // Don't use 'BuildContext's across async gaps.
        break;
    }
  }
}
```

**Expected behavior**
`use_build_context_synchronously` lint works fine.

**Comment by arnovanliere** [comment]
Possibly related to #4703?

**Comment by maRci002** [comment]
> Possibly related to #4703?

The issue with `!context.mounted` has always been a problem for me, but the issue with the switch statement began when I upgraded to Flutter 3.13 and Dart 3.1.0. Essentially, after upgrading Flutter, I received numerous warnings in my codebase.

It's worth noting that when I move the first case containing await to a position below the other cases, the problem goes away.

```dart
import 'package:flutter/material.dart';

Future<void> test(BuildContext context) async {
  if (context.mounted) {
    switch (1) {
      case 2:
        Theme.of(context); // Works fine
        break;
      case 1:
        await Future<void>.delayed(Duration.zero);
        break;
    }
  }
}
```

**Comment by srawlins** [comment]
Fixed in https://github.com/dart-lang/linter/commit/411a3f5e76f2a00c43ad90491805b71db81d7ee2, will ship with Dart 3.2.0. Apologies for the delay.
